firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleSerializer.java

```diff
@@ -61,7 +80,7 @@ public NamedQuery decodeNamedQuery(String json) throws JSONException {
   public NamedQuery decodeNamedQuery(JSONObject namedQuery) throws JSONException {
     String name = namedQuery.getString("name");
     BundledQuery bundledQuery = decodeBundledQuery(namedQuery.getJSONObject("bundledQuery"));
-    SnapshotVersion readTime = decodeSnapshotVersion(namedQuery.getJSONObject("readTime"));
+    SnapshotVersion readTime = decodeSnapshotVersion(namedQuery.getString("readTime"));
```

Comment:
Do we need to support both? That is, is it ever possible that users might end up with timestamp values encoded as a seconds/nanoseconds pair instead?

I'm specifically concerned because `Timestamp.toJSON` uses the object format: https://github.com/firebase/firebase-js-sdk/blob/f68bd3f29c87e98f15635e38c1f9847b7731fa30/packages/firestore/src/lite/timestamp.ts#L162. It seems that if users construct bundles for themselves they could get this wrong.

Reply:
That is to say, it seems like logic like we have in the JS SDK is required for dealing with JSON timestamps, no?

https://github.com/firebase/firebase-js-sdk/blob/f68bd3f29c87e98f15635e38c1f9847b7731fa30/packages/firestore/src/model/normalize.ts#L31

Reply:
I updated this PR to support both formats. I am not sure we need this going forward, but this ensures backwards compatibility with existing bundles.